include/wil/filesystem.h

```diff
@@ -1043,8 +1046,224 @@ namespace wil
         THROW_IF_FAILED(GetFileInfoNoThrow<infoClass>(fileHandle, result));
         return result;
     }
+
+    // Helpers to make the CreateFileW API easier to use. This segregates the OPEN_EXISTING "open an existing file"
+    // cases from the "create a new file" that has 4 variations represented in the create_file_create_behavior enum.
+
+    // https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfilew?devlangs=cpp&f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(FILEAPI%252FCreateFileW)%3Bk(CreateFileW)%3Bk(DevLang-C%252B%252B)%3Bk(TargetOS-Windows)%26rd%3Dtrue
+
+    struct file_and_error_result

```

Comment:
Maybe I'm just weird, but I'd be fine with even the `try_` versions returning a `wil::unique_handle` with the expectation that the user queries it for validity (& then calls `GetLastError` if desired). That would be consistent with how other "try" functions tend to work such as `wil::com_ptr_t::try_query`. The usage is also not _that_ differnet. E.g. consider the two possible use cases:
```c++
// CURRENT
auto [handle, err] = wil::try_open_file(path);
RETURN_IF_WIN32_ERROR(err);

// ALTERNATIVE
auto handle = wil::try_open_file(path);
RETURN_LAST_ERROR_IF(!handle);
```
The usage is pretty similar. It might be slightly more work if someone wants to change behavior based on the specific error, but that's probably not the most common use.

Reply:
I want to provide the last error in success cases too, so that motivated me to use the struct. if you still disagree let me know and I can reconsider.

Reply:
I still think that _most_ consumers don't need/care about the last set error value on success. Or at least I've never personally called `GetLastError` when I've gotten back a valid handle. That said, I was mostly just bringing it up for discussion; I'm okay with it either way.

Reply:
Another data point/thing to consider: the exception versions don't communicate the error code